### PR TITLE
v2.3.0 release prep

### PR DESCRIPTION
**Pull request by shuchitak**
- Add doc build
- Update version to v2.3.0
- Fix headings in README.rst

### CODE REVIEW

**Review by ed-xmos — COMMENTED** [review]
As before - minor release I think so 2.3.0

File: doc/rst/locks.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+################################
+lib_locks: Lock handling Library
```
Comment (xross) [inline doc/rst/locks.rst:2]: Capitalisation issue

Comment (xross) [reply doc/rst/locks.rst:2]: Or should Library be capitalised?

File: settings.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+lib_name: lib_locks
+project: '{{lib_name}}'
+title: '{{lib_name}}: Lock handling library'
+version: 4.3.0
```
Comment (xross) [inline settings.yml]:
```suggestion
version: 2.3.0
```

Comment (ed-xmos) [reply settings.yml]: That's a very major version bump!

File: doc/rst/locks.rst (hunk 1 above)
Comment (shuchitak) [reply doc/rst/locks.rst:2]: I think handling needs capitalisation? looking at other examples...

File: settings.yml (hunk 2 above)
Comment (shuchitak) [reply settings.yml]: Oops! Sorry! Everything is starting to look the same at this point..

**Review by xross — APPROVED** [review]
Looks good

**Pull request merged**